lib/mkFakeDerivation.nix

```diff
@@ -9,10 +12,47 @@
       meta = {outputsToInstall = ["out"];} // (publishData.eval.meta or {});
       outputs = publishData.eval.outputs or {"out" = lib.head publishData.element.storePaths;};
     };
-  fromSource = with publishData.element;
-    if url == ""
-    then throw "url = \"\" so this fakeDerivation can't be built from source. Note that fakeDerivations created from self cannot be built from source"
-    else lib.getAttrFromPath attrPath (builtins.getFlake url);
+  fromSource = let
+    fromNixpkgs = builtins.match "github:flox/nixpkgs/.*" publishData.element.url != null;
+    # implement allowedUnfreePackages as allowUnfreePredicate, but still honor
+    # allowUnfreePredicate
+    allowUnfreePredicate =
+      if context ? config.nixpkgs-config.allowUnfreePredicate
+      then
+        if context ? config.nixpkgs-config.allowedUnfreePackages
+        then throw "only one of allowedUnfreePackages and allowUnfreePredicate can be specified in config.nixpkgs-config"
+        else context.config.nixpkgs-config.allowUnfreePredicate
+      else if context ? config.nixpkgs-config.allowedUnfreePackages
+      then pkg: builtins.elem (lib.getName pkg) context.config.nixpkgs-config.allowedUnfreePackages
```

Comment:
Mind, `getName` is not necessarily unique

adding `"myUnfreeThing"` allows all of `myUnfreeThing`,  `pythonPackages.myUnfreeThing`,  `nodePackages.myUnfreeThing` etc,..
maybe that's desired, maybe its not...

what it means is that i can taken any unfree package add an alias of it in `unlockme.myUnfreeThing` and be able to build it.

Reply:
Yeah that's just conventional upstream. No reason we can't do something better

Reply:
Pushed a fix